src/Bundle/spec/Form/DataTransformer/ResourceToIdentifierCacheableTransformerSpec.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ResourceBundle\Form\DataTransformer;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+final class ResourceToIdentifierCacheableTransformerSpec extends ObjectBehavior
+{
+    function let(RepositoryInterface $repository): void
+    {
+        $this->beConstructedWith($repository, 'id');
+    }
+
+    function it_does_not_reverses_null_value(RepositoryInterface $repository): void
+    {
+        $repository->findOneBy(Argument::any())->shouldNotBeCalled();
+
+        $this->reverseTransform(null)->shouldReturn(null);
+    }
+
+    function it_throws_an_exception_on_non_existing_resource(RepositoryInterface $repository): void
+    {
+        $repository->getClassName()->willReturn(ResourceInterface::class);
+        $repository->findOneBy(['id' => 6])->willReturn(null);
+
+        $this->shouldThrow(TransformationFailedException::class)->during('reverseTransform', [6]);
+    }
+
+    function it_reverse_transform_identifier_to_resource(RepositoryInterface $repository, ResourceInterface $resource): void
+    {
+        $repository->findOneBy(['id' => 5])->willReturn($resource);
+
+        $this->reverseTransform(5)->shouldReturn($resource);
+    }
+
+    function it_reverse_transforms_identifier_to_resource_using_cache(
+        RepositoryInterface $repository,
+        ResourceInterface $resource,
+    ): void {
+        $this->clear();
```

Comment:
Creating a method just for the specs to work is a bad idea. Even more so since it's not in any interface. Normally for something like this, you could use `ResetInterface`, but in this particular case, it's useless since this is not a service living in the container and instead gets recreated all the time, it'd be confusing.

Remove the `::clear`, and as for the test itself, you can use [reflection](https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Bundle/CoreBundle/spec/Theme/ChannelBasedThemeContextSpec.php#L67) to set the value of cache